aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -268,5 +948,88 @@ Tensor replication_pad1d_backward_cpu(
   return gradInput;
 }
 
+Tensor& replication_pad2d_out_cpu(
+    Tensor& output,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  replication_pad2d_out_cpu_template(
+      output, input, paddingSize);
+  return output;
+}
+
+Tensor replication_pad2d_cpu(
+    const Tensor& input,
+    IntList paddingSize)
+{
+  auto output = at::empty({0}, input.options());
+  replication_pad2d_out_cpu_template(
+      output, input, paddingSize);
+  return output;
+}
+
+Tensor& replication_pad2d_backward_out_cpu(
+    Tensor& gradInput,
+    const Tensor& gradOutput,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  replication_pad2d_backward_out_cpu_template(
+      gradInput, gradOutput, input, paddingSize);
+  return gradInput;
+}
+
+Tensor replication_pad2d_backward_cpu(
+    const Tensor& gradOutput,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  auto gradInput = at::zeros_like(input);
+  replication_pad2d_backward_out_cpu_template(
+      gradInput, gradOutput, input, paddingSize);
+  return gradInput;
+}
+
+Tensor& replication_pad3d_out_cpu(
+    Tensor& output,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  replication_pad3d_out_cpu_template(
+      output, input, paddingSize);
+  return output;
+}
+
+Tensor replication_pad3d_cpu(
+    const Tensor& input,
+    IntList paddingSize)
+{
+  auto output = at::empty({0}, input.options());
+  replication_pad3d_out_cpu_template(
+      output, input, paddingSize);
+  return output;
+}
+
+Tensor& replication_pad3d_backward_out_cpu(
+    Tensor& gradInput,
+    const Tensor& gradOutput,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  replication_pad3d_backward_out_cpu_template(
+      gradInput, gradOutput, input, paddingSize);
+  return gradInput;
+}
+
+Tensor replication_pad3d_backward_cpu(
+    const Tensor& gradOutput,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  auto gradInput = at::zeros_like(input);
```

Comment:
IIUC, you're going to zero gradInput inside the function anyway, so no need to zero it here. `empty_like` would be sufficient.